impl/src/main/java/impl/QuestionResourceImpl.java

```diff
@@ -119,13 +127,32 @@ public QuestionResourceImpl(QuestionDao questionDao, QuestionVoteDao questionVot
                 question.setId(longGeneratedKey.getKey());
                 return question;
             })
-            .flatMap(savedQuestion -> {
-                Observable<Void> postMessageToSlack = slackResource.postMessageToSlack(slackConfig.getFeedChannel(), notificationMessage(question))
+            .concatMap(savedQuestion -> {
+                Observable<Void> operations = slackResource.postMessageToSlack(slackConfig.getFeedChannel(), notificationMessage(question))
                     .onErrorResumeNext(e -> {
                         LOG.error("failed to notify by slack that question has been added", e);
                         return empty();
                     });
-                return first(postMessageToSlack).thenReturn(savedQuestion);
+
+                if(question.getTags() != null) {
+                    List<String> lowerCasedTags = question
```

Comment:
extract into methods, to reduce brain overflow

Reply:
I did a best effort to extract a method. If we see that it can be improved we could perhaps mob it?